src/api/verification_case.py

```diff
@@ -0,0 +1,430 @@
+                                logging.error(
+                                    f"The type of '{idf_key}' value must be either float or int, but {type(instance[key][idf_key])} is provided."
+                                )
+                                return False
+                            else:
+                                if verbose:
+                                    print(
+                                        f"The correct {idf_key} type {type(idf_value)} is provided."
+                                    )
+
+            return is_cases_valid
+
+        # check case type
+        if not isinstance(case, Dict):
+            logging.error(
+                f"The case argument type must be dict, but {type(case)} is provided."
+            )
+            return None
+
+        # check verbose type
+        if not isinstance(verbose, bool):
+            logging.error(
+                f"The verbose argument type must be bool, but {type(verbose)} is provided."
+            )
+            return None
+
+        case_schema = {
+            "no": int,
+            "run_simulation": bool,
+            "simulation_IO": {
+                "idf": str,
+                "idd": str,
+                "weather": str,
+                "output": str,
+                "ep_path": str,
+            },
+            "expected_result": str,
+            "datapoints_source": {"idf_output_variables": dict, "parameters": dict},
+            "verification_class": str,
+        }
+        return _validate_case_structure_helper(case_schema, case, verbose)
+
+    @staticmethod
+    def save_verification_cases_to_json(
+        json_path: str = None, cases: list = None
+    ) -> NoReturn:
+        """Save verification cases to a dedicated file. The cases list consists of verification case dicts.
+
+        Args:
+            json_path: str. json file path to save the cases.
+            cases: List. List of complete verification cases Dictionary to save.
+        """
+        # check `json_path` type
+        if not isinstance(json_path, str):
+            logging.error(
+                f"The `json_path` argument type must be str, but {type(json_path)} is provided."
+            )
+            return None
+
+        # check `cases` type
+        if not isinstance(cases, list):
+            logging.error(
+                f"The `cases` argument type must be list, but {type(cases)} is provided."
+            )
+            return None
+
+        # check if `json_path` extension is .json
+        if not json_path[-5:] == ".json":
+            logging.error(f"The `json_path` argument must end with '.json' extension.")
+            return None
+
+        # organize the cases in the correct format
+        case_suite_in_template_format = {"cases": []}
+        case_suite_in_template_format_append = case_suite_in_template_format[
+            "cases"
+        ].append
+        for case in cases:
+            case_suite_in_template_format_append(case)
```

Comment:
feels unnecessary, just:
```suggestion
        case_suite_in_template_format = {"cases": cases}
```

Reply:
great point! Fixed.